### DISCUSSION TITLE
Can't upload media to google photos

### DISCUSSION THREAD

**Discussion by orfest** [Byron/google-apis-rs#310] [Q&A]
Hello,

I've spent a couple of hours looking at the docs of the rust package, but didn't find a way to upload bytes to the Google Photos server.
Maybe I'm missing something obvious, but I didn't manage to find any references to the `/v1/uploads` method.
Can you please add an example of how to upload bytes?

**Comment by Byron** [comment]
The Rust crate is entirely generated from their API spec, which doesn't constitute great documentation unfortunately. It also links to [the official docs](https://developers.google.com/photos/library/guides/upload-media?hl=en) which generally is much more helpful for getting a bearing.

With that knowledge, it should be possible to go back to the crate and find the calls you are looking for. Uploads are generally supported [and somewhat described here](https://docs.rs/google-photoslibrary1/2.0.4+20210331/google_photoslibrary1/#uploads-and-downloads).

> Can you please add an example of how to upload bytes?

Please consult the official google docs for examples and guides, this project is just a binding generator.

**Comment by orfest** [comment]
I `grep`ped all files in the `gen` folder, and in the `src` folder. The only `upload` function is in `struct ResumableUploadHelper`. I could make use of this struct directly, but that defeats the purpose of having an abstraction layer at all.

**Comment by orfest** [comment]
> It should be possible.

Maybe it is possible. I actually spent two hours trying to make it happen, and couldn't. Please give me a at least a hint where to look.
Also your answer assumes that the generator covers all possible methods. Is there a possibility that the code for `/v1/uploads` wasn't generated?

**Comment by Byron** [comment]
Based on what I see in the [upload media docs](https://developers.google.com/photos/library/guides/upload-media) I think the first step is to upload bytes. For that I don't see any generated methods as it's apparently outside of the specification or the generator can't pick it up.

This is probably something to do by hand using an HTTP client, and take it from there with the creation of media items, for which there is an API surface.
